### PR TITLE
Created User Actions/Reducers

### PR DESCRIPTION
**Pull request by nateboyette**
* Created User Actions and Reducers. 
* Some actions were written but not fully implemented since they may not be needed. 
* Flow type structure can be used as a template for all other actions and reducers.

### CODE REVIEW

File: src/store/actions/userActions.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    );
+
+    // purposely left disconnected from user reducer until/if needed
+  } catch (error) {
+    dispatch({ type: GET_USER_DATA_FAIL });
+  }
+};
+
+export const editUserData = (
+  changes: ?Object,
```
Comment (Alex-AK) [inline src/store/actions/userActions.js]: ? is not needed.

File: src/auth/react-auth0-wrapper.jsx [hunk 2]
```diff
@@ -1,10 +1,18 @@
 // src/react-auth0-wrapper.js
 import React, { useState, useEffect, useContext } from 'react';
 import createAuth0Client from '@auth0/auth0-spa-js';
-import axios from 'axios';
 import jwt from 'jsonwebtoken';
 import Axios from '../utils/axios_config';
 
+// state management
+import store from '../store/store';
+import { getUserByEmail } from '../store/actions/userActions';
```
Comment (Alex-AK) [inline src/auth/react-auth0-wrapper.jsx]: I don't think we're using this import. Could be removed?

File: src/store/actions/userActions.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  dispatch: Dispatch<UserActionType, Object>
+) => {
+  dispatch({ type: EDIT_USER_DATA_START });
+
+  try {
+    await Axios.put(`/users/${userId}`, changes, {
+      headers: { Authorization: token }
+    });
+
+    dispatch({ type: EDIT_USER_DATA_SUCCESS, payload: changes });
```
Comment (Alex-AK) [inline src/store/actions/userActions.js]: I typically pull the data out of the response and use it as a conditional to confirm before dispatching a success so the database and redux store are synced. 

Something like 
`const {data} = await Axios.put`
`if (data) dispatch(...)`

File: src/store/actions/userActions.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  dispatch: Dispatch<UserActionType, Object>
+) => {
+  dispatch({ type: DELETE_USER_START });
+
+  try {
+    await Axios.delete(`/users/${userId}`, {
+      headers: { Authorization: token }
+    });
+
+    dispatch({ type: DELETE_USER_SUCCESS, payload: userId });
```
Comment (Alex-AK) [inline src/store/actions/userActions.js]: Same concept here for dispatching store changes.

File: src/store/actions/userActions.js (hunk 3 above)
Comment (nateboyette) [reply src/store/actions/userActions.js]: This endpoint returns a response code but doesn't return any data at the moment. Do you want an additional check for '200' before dispatching?

**Pull request merged**